api/server/router/build/build.go

```diff
@@ -3,19 +3,23 @@ package build
 import (
 	"github.com/docker/docker/api/server/router"
 	"github.com/docker/docker/api/server/router/local"
-	"github.com/docker/docker/daemon"
+	"github.com/docker/docker/builder"
 )
 
 // buildRouter is a router to talk with the build controller
 type buildRouter struct {
-	backend *daemon.Daemon
-	routes  []router.Route
+	// There are two backends for build endpoints.
+	// apiBackend implements the api endpoint for /build.
+	// buildBackend implements the builder's Backend interface.
+	apiBackend   Backend
+	buildBackend builder.Backend
```

Comment:
As discussed live, this should be moved down to the `Build()` implementation so that the API doesn't depend on `builder`. PR will be updated accordingly.